staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -0,0 +1,378 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	apps "k8s.io/api/apps/v1beta1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/informers"
+	appsinformers "k8s.io/client-go/informers/apps/v1beta1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	example "k8s.io/sample-controller/pkg/apis/example/v1alpha1"
+	"k8s.io/sample-controller/pkg/client/clientset/versioned"
+	intinformers "k8s.io/sample-controller/pkg/client/informers/externalversions"
+	exampleinformers "k8s.io/sample-controller/pkg/client/informers/externalversions/example/v1alpha1"
+)
+
+type Controller struct {
+	// clientset is a standard kubernetes clientset
+	clientset kubernetes.Interface
+	// exampleclientset is a clientset for our own API group
+	exampleclientset versioned.Interface
+
+	// informerFactory produces shared informers for kubernetes types
+	informerFactory informers.SharedInformerFactory
```

Comment:
would also prefix this with kube

Reply:
the informer factories shouldn't be part of this struct, but instatiated and started outside. They are meant to be shared if you have multiple controllers per binary.